### PR TITLE
⬆️ Maintenance upgrades tests and tools libraries

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Updates tests and tooling libraries *before updating services*. Since it was a while since last update, there are many changes. 
Nonetheless, i call your attention on the **major** updates, namely:
- pytest 8: https://docs.pytest.org/en/stable/changelog.html#breaking-changes
- dask-family (only testing)
- black (tooling)
- cryptograph (only testing)
- docker (only testing)
- httpcore (only testing)
- watchdog (only testing)

### Changes in code

- ♻️ ruffed all old `setup.py` 
- ⬆️ api/test: removed deprecated `openapi_spec_validator.validate_spec` 
- 🎨 bad tests in `test_httpx_utils.py` that hard-coded version
- 🗑️ `pytest-lazyfixture` fails with pytest 8 (SEE https://github.com/TvoroG/pytest-lazy-fixture/issues/65): Modified the tests code to use parametrized fixtures instead of lazy-fixtures and **removed the dependency.**
- 🎨 uses `pytest-asyncio` (constraint to <0.22 due to https://github.com/pytest-dev/pytest-asyncio/issues/706). 
- 🗑️ Removes `pytest-aiohttp` where is not needed. NOTE: it is necessary until `aiohttp.pytest_plugin.loop` fixture retired!
- `aiohttp<3.9`: Changes in 3.9 needs to be addressed on a separate PR
- ♻️ migration scripts in tests to get it to work after the update!

### In detail

|#|name|before|after|upgrade|count|packages|
|-|----|------|-----|-------|-----|--------|
|   1 | aiohttp                   | 3.8.5, 3.8.6, 3.9.1 | 3.8.6      | 🔥 downgrade | 7 | aws-library🧪</br>dask-sidecar🧪</br>dask-task-models-library🧪</br>environment-setup🧪</br>models-library🧪</br>osparc-gateway-server🧪</br>postgres-database🧪 |
|   2 | arrow                     | 1.3.0           | 🗑️ removed |  | 3 | environment-setup🧪</br>service-library🧪🧪 |
|   3 | dnspython                 | 2.4.2           | 🗑️ removed |  | 1 | service-library🧪 |
|   4 | email-validator           | 2.1.0.post1     | 🗑️ removed |  | 1 | service-library🧪 |
|   5 | idna                      | 3.6, 3.4        | 3.4        | 🔥 downgrade | 4 | dask-task-models-library🧪</br>models-library🧪</br>osparc-gateway-server🧪</br>postgres-database🧪 |
|   6 | jsonschema                | 4.19.2          | 🗑️ removed |  | 2 | environment-setup🧪</br>service-library🧪 |
|   7 | jsonschema-specifications | 2023.7.1        | 🗑️ removed |  | 2 | environment-setup🧪</br>service-library🧪 |
|   8 | markdown-it-py            | 3.0.0           | 🗑️ removed |  | 3 | environment-setup🧪</br>service-library🧪🧪 |
|   9 | mdurl                     | 0.1.2           | 🗑️ removed |  | 3 | environment-setup🧪</br>service-library🧪🧪 |
|  10 | orjson                    | 3.9.10          | 🗑️ removed |  | 3 | environment-setup🧪</br>service-library🧪🧪 |
|  11 | pydantic                  | 1.10.13         | 🗑️ removed |  | 1 | service-library🧪 |
|  12 | pygments                  | 2.16.1          | 🗑️ removed |  | 3 | environment-setup🧪</br>service-library🧪🧪 |
|  13 | pytest-aiohttp            | 1.0.5           | 🗑️ removed |  | 10 | aws-library🧪</br>dask-sidecar🧪</br>dask-task-models-library🧪</br>director-v2🧪</br>environment-setup🧪</br>models-library🧪</br>postgres-database🧪</br>simcore-sdk🧪</br>swarm-deploy🧪</br>tests🧪 |
|  14 | pytest-asyncio            | 0.21.1, 0.23.5, 0.23.2 | 0.21.1     | 🔥 downgrade | 6 | autoscaling🧪</br>aws-library🧪</br>dask-task-models-library🧪</br>notifications-library🧪</br>osparc-gateway-server🧪</br>public-api🧪 |
|  15 | pytest-lazy-fixture       | 0.6.3           | 🗑️ removed |  | 1 | simcore-sdk🧪 |
|  16 | referencing               | 0.29.3          | 🗑️ removed |  | 2 | environment-setup🧪</br>service-library🧪 |
|  17 | rich                      | 13.6.0          | 🗑️ removed |  | 3 | environment-setup🧪</br>service-library🧪🧪 |
|  18 | rpds-py                   | 0.10.6          | 🗑️ removed |  | 2 | environment-setup🧪</br>service-library🧪 |
|  19 | typer                     | 0.9.0           | 🗑️ removed |  | 3 | environment-setup🧪</br>service-library🧪🧪 |
|  20 | types-python-dateutil     | 2.8.19.14       | 🗑️ removed |  | 3 | environment-setup🧪</br>service-library🧪🧪 |
|  21 | yarl                      | 1.9.4, 1.9.2    | 1.9.2      | 🔥 downgrade | 7 | aws-library🧪</br>dask-sidecar🧪</br>dask-task-models-library🧪</br>environment-setup🧪</br>models-library🧪</br>osparc-gateway-server🧪</br>postgres-database🧪 |

*Legend*: 
- ⬆️ base dependency (only services because packages are floating)
- 🧪 test dependency
- 🔧 tool dependency

### CODE REVIEW

**Review by bisgaard-itis — APPROVED** [review]
Thanks a lot for the effort

**Review by matusdrobuliak66 — APPROVED** [review]
thanks :100:

**Review by GitHK — COMMENTED** [review]
please ask me for a review when you mange to get the tests green again

File: requirements/constraints.txt [hunk 1]
```diff
@@ -82,3 +86,6 @@ aiofile>=999999999
 # directly use http calls.
 # SEE https://github.com/Pennsieve/pennsieve-python/issues/17
 pennsieve>=999999999
+
+# User alternative e.g. parametrized fixture or request.getfixturevalue(.)
+pytest-lazy-fixture>=999999999
```
Comment (GitHK) [inline requirements/constraints.txt:91]: I guess you needed it here to upgrade and remove it.
Is it still required?

**Review by GitHK — APPROVED** [review]
Thanks for all this. I see we are getting behind on upgrades and need to address all this

File: requirements/constraints.txt (hunk 1 above)
Comment (pcrespov) [reply requirements/constraints.txt:91]: @GitHK this is how you avoid this library to be ever installed here. 
image

**Pull request merged**